src/module/actor/sheet/base.js

```diff
@@ -395,6 +396,89 @@ export class ActorSheetSFRPG extends ActorSheet {
         }
     }
 
+    _prepareAttackString(item)  {
+        try {
+            const itemData = item.system;
+            const actor = item.actor;
+            const isWeapon = ["weapon", "shield"].includes(item.type);
+
+            let abl = itemData.ability;
+            if (!abl && (actor.type === "npc" || actor.type === "npc2")) abl = "";
+            else if (!abl && (item.type === "spell")) abl = actor.system.attributes.spellcasting || "int";
+            else if (itemData.properties?.operative && actor.system.abilities.dex.value > actor.system.abilities.str.value) abl = "dex";
+            else if (!abl) abl = "str";
+
+            // Define Roll parts
+            const parts = [];
+
+            if (Number.isNumeric(itemData.attackBonus) && itemData.attackBonus !== 0) parts.push("@item.attackBonus");
+            if (abl) parts.push(`@abilities.${abl}.mod`);
+            if (["character", "drone"].includes(actor.type)) parts.push("@attributes.baseAttackBonus.value");
+            if (isWeapon) {
+                const procifiencyKey = SFRPG.weaponTypeProficiency[item.system.weaponType];
+                const proficient = itemData.proficient || actor?.system?.traits?.weaponProf?.value?.includes(procifiencyKey);
+                if (!proficient) {
+                    parts.push(`-4[${game.i18n.localize("SFRPG.Items.NotProficient")}]`);
+                }
+            }
+
+            const formula = parts.join("+");
+
+            const appropriateMods = item.getAppropriateAttackModifiers(isWeapon);
+            const stackModifiers = new StackModifiers();
+            let modifiers = stackModifiers.process(appropriateMods, null);
```

Comment:
If you want to have this working with #880 please make sure that you do NOT include formula modifiers as they will throw errors when pushed to the non async `stackmodifiers.process()` or at least will not work. Same applies for `_prepareDamageString()` or any other usage of `stackmodifiers.process()`.

Reply:
`getAppropriateAttackModifiers` and `getAppropriateDamageModifiers` only return constant modifiers, not situational modifiers. They can include things like `@details.level.value`. Will that be alright?